frontend/src/pages/exploreApplication/EnableModal.tsx

```diff
@@ -53,31 +54,58 @@ const EnableModal: React.FC<EnableModalProps> = ({ selectedApp, shown, onClose }
   };
 
   React.useEffect(() => {
-    if (validationInProgress && validationStatus === EnableApplicationStatus.SUCCESS) {
-      setValidationInProgress(false);
-      // TODO: Disable rule below temporarily. Refactor to notify the owner and avoid modifying the object directly.
-      /* eslint-disable no-param-reassign */
-      selectedApp.spec.isEnabled = true;
-      selectedApp.spec.shownOnEnabledPage = true;
-      /* eslint-enable no-param-reassign */
+    if (
+      validationInProgress &&
+      validationStatus === EnableApplicationStatus.SUCCESS &&
+      selectedApp.spec.internalRoute
+    ) {
```

Comment:
I think this would break functionality for non internalRoute apps

So you may need to give a way for them to handle the success too like
```suggestion
    if (
      validationInProgress &&
      validationStatus === EnableApplicationStatus.SUCCESS
    ) {
      if (selectedApp.spec.internalRoute) {...your changes}
      else { ...old code}
```